### DISCUSSION TITLE
Moving grouped tabs breaks the grouping

### DISCUSSION THREAD

**Discussion by rocketeeer** [Able-Bear-Software/tab-nodes-tree#56] [Ideas / Feature Request]
Version: 1.4.0

Issue: Moving grouped tabs appears to remove the top level hierarchy relationship and move all child tabs up one level of tree hierarchy.  Also the child groups seem to be expanded without the expand/shrink toggle changing.

Steps to reproduce:
1. Create a new tab (referred to below as the 'parent' tab}
2. Create one or more child tabs under the parent tab (referred to below as 'first' children tabs)
3. Create one or more tabs under the first children tabs
4. Repeat step above until you are bored
5. Collapse the tab group so only the parent tab is visible
6. Drag the parent tab up or down the list of tabs

Expected Results: Ideally grouped tabs would remain grouped while repositioning the group.

Video attached below

https://github.com/user-attachments/assets/c4449f9c-64f2-491f-b4f3-5fbf0fe28dc8

**Comment by zeroarst** [comment]
Hi @rocketeeer,

Hmm, interesting! I think the key question here is: When moving a collapsed subtree, should the whole subtree (including all descendants) move with it? I feel like it should be expected behavior like you mentioned. However, for an expanded subtree, moving the parent tab node should not move the descendants.

As for the collapsed subtree expanding after moving the parent tab node, this is necessary; otherwise, the subtree would essentially disappear.

By the way, I love how you named the tabs for such a clear demonstration—well done! 🤟

P.S. I use "collapsed/expanded" terms to differentiate from the native "grouping" function.

Ticked created.  #57

**Comment by zeroarst** [comment]
https://github.com/Able-Bear-Software/tab-nodes-tree/discussions/59#discussioncomment-10993835

**Comment by zeroarst** [comment]
Added to [1.4.1](https://github.com/Able-Bear-Software/tab-nodes-tree/discussions/59).
